psi4/src/psi4/libsapt_solver/usapt0.cc

```diff
@@ -0,0 +1,2459 @@
+    outfile->Printf( "    %-6s %5d %5d\n", "nvir", eps_avira_A_->dim() + eps_fvira_A_->dim(), eps_avira_B_->dim() + eps_fvira_B_->dim());
+    outfile->Printf( "    %-6s %5d %5d\n", "nfocc", eps_focca_A_->dim(), eps_focca_B_->dim());
+    outfile->Printf( "    %-6s %5d %5d\n", "naocc", eps_aocca_A_->dim(), eps_aocca_B_->dim());
+    outfile->Printf( "    %-6s %5d %5d\n", "navir", eps_avira_A_->dim(), eps_avira_B_->dim());
+    outfile->Printf( "    %-6s %5d %5d\n", "nfvir", eps_fvira_A_->dim(), eps_fvira_B_->dim());
+    outfile->Printf( "    ------------------\n");
+    outfile->Printf( "      Beta orbitals  \n");
+    outfile->Printf( "    ------------------\n");
+    outfile->Printf( "    %-6s %5d %5d\n", "nmo", nmob_A, nmob_B);
+    outfile->Printf( "    %-6s %5d %5d\n", "nocc", eps_aoccb_A_->dim() + eps_foccb_A_->dim(), eps_aoccb_B_->dim() + eps_foccb_B_->dim());
+    outfile->Printf( "    %-6s %5d %5d\n", "nvir", eps_avirb_A_->dim() + eps_fvirb_A_->dim(), eps_avirb_B_->dim() + eps_fvirb_B_->dim());
+    outfile->Printf( "    %-6s %5d %5d\n", "nfocc", eps_foccb_A_->dim(), eps_foccb_B_->dim());
+    outfile->Printf( "    %-6s %5d %5d\n", "naocc", eps_aoccb_A_->dim(), eps_aoccb_B_->dim());
+    outfile->Printf( "    %-6s %5d %5d\n", "navir", eps_avirb_A_->dim(), eps_avirb_B_->dim());
+    outfile->Printf( "    %-6s %5d %5d\n", "nfvir", eps_fvirb_A_->dim(), eps_fvirb_B_->dim());
+    outfile->Printf( "    ------------------\n");
+    outfile->Printf( "\n");
+
+    outfile->Printf( "   => Primary Basis Set <=\n\n");
+    primary_->print_by_level("outfile", print_);
+
+}
+
+void USAPT0::print_trailer()
+{
+    energies_["delta HF,r (2)"] = 0.0;
+    energies_["delta HF,u (2)"] = 0.0;
+    if (energies_["HF"] != 0.0) {
+        if (coupled_ind_) {
+          energies_["delta HF,r (2)"] = energies_["HF"] - energies_["Elst10,r"] - energies_["Exch10"] - energies_["Ind20,r"] - energies_["Exch-Ind20,r"];
+        } else {
+          energies_["delta HF,u (2)"] = energies_["HF"] - energies_["Elst10,r"] - energies_["Exch10"] - energies_["Ind20,u"] - energies_["Exch-Ind20,u"];
+        }
+    }
+
+    energies_["Electrostatics"] = energies_["Elst10,r"];
+    energies_["Exchange"]       = energies_["Exch10"];
+    if (coupled_ind_) {
+        energies_["Induction"]      = energies_["Ind20,r"] + energies_["Exch-Ind20,r"] + energies_["delta HF,r (2)"];
+    } else {
+        energies_["Induction"]      = energies_["Ind20,u"] + energies_["Exch-Ind20,u"] + energies_["delta HF,u (2)"];
+    }
+    energies_["Dispersion"]     = energies_["Disp20"] + energies_["Exch-Disp20"];
+    energies_["SAPT"]           = energies_["Electrostatics"] + energies_["Exchange"] + energies_["Induction"] + energies_["Dispersion"];
+
+    outfile->Printf("\n    SAPT Results  \n");
+    outfile->Printf("  -----------------------------------------------------------------------\n");
+    outfile->Printf("    Electrostatics     %16.8lf mH %16.8lf kcal mol^-1\n",
```

Comment:
Output now exactly matches SAPT0 output formatting. Scaling factor included too, as an option, but sSAPT0 is not printed out. We only recommend it for closed-shells since it was never tested on open-shell...